cosign/verifier.go

```diff
@@ -0,0 +1,544 @@
+	sigstoreBundleMediaType01 = "application/vnd.dev.sigstore.bundle+json;version=0.1"
+	bundleAnnotationKey       = "dev.sigstore.cosign/bundle"
+	certKey                   = "dev.sigstore.cosign/certificate"
+	signatureAnnotationKey    = "dev.cosignproject.cosign/signature"
+	simpleSigningMediaType    = "application/vnd.dev.cosign.simplesigning.v1+json"
+)
+
+// PublicKeyConfig contains the configuration for key-based verification.
+type PublicKeyConfig struct {
+	// PublicKey for key-based verification.
+	PublicKey crypto.PublicKey
+
+	// SignatureAlgorithm defines the algorithm used for signature verification.
+	// Optional. If not provided, defaults to SHA256.
+	SignatureAlgorithm crypto.Hash
+
+	// ValidityPeriodStart defines the start time for the public key validity
+	// period. Optional. If not provided, the key is considered valid from the
+	// beginning of time.
+	ValidityPeriodStart time.Time
+
+	// ValidityPeriodEnd defines the end time for the public key validity period.
+	// Optional. If not provided, the key is considered valid until the end of
+	// time.
+	ValidityPeriodEnd time.Time
+}
+
+// VerifierOptions contains the options for creating a new Cosign verifier.
+type VerifierOptions struct {
+	// Name is the instance name of the verifier to be created. Required.
+	Name string
+
+	// TrustedRoot defines the trusted root for verification. For keyless verification,
+	// this will contain Fulcio certificates and Rekor public keys. Optional.
+	// If not provided, the default Sigstore trusted root will be used.
+	TrustedRoot *root.TrustedRoot
+
+	// PublicKeyConfig contains the public key and its validity period for key-based verification.
+	// If provided, key-based verification will be performed. Optional.
+	PublicKeyConfigs []*PublicKeyConfig
+
+	// IdentityPolicies contains policies for keyless verification.
+	// These policies specify which OIDC identities are trusted. Optional.
+	IdentityPolicies []verify.PolicyOption
+
+	// IgnoreTlog when set to true, skips Artifact transparency log verification.
+	// Only applies to keyless verification. Optional, defaults to false.
+	IgnoreTlog bool
+
+	// IgnoreCTLog when set to true, skips certificate transparency log verification.
+	// Only applies to keyless verification. Optional, defaults to false.
+	IgnoreCTLog bool
+
+	// TUFOptions provides custom TUF client options for fetching trusted root.
+	// Optional.
+	TUFOptions *tuf.Options
+}
+
+// Verifier is a ratify.Verifier implementation that verifies Cosign signatures.
+type Verifier struct {
+	name             string
+	identityPolicies []verify.PolicyOption
+	ignoreTlog       bool
+	verifier         *verify.Verifier
+}
+
+// NewVerifier creates a new Cosign verifier.
+func NewVerifier(opts *VerifierOptions) (*Verifier, error) {
+	if opts.Name == "" {
+		return nil, fmt.Errorf("verifier name is required")
+	}
+
+	var trustedMaterial = make(root.TrustedMaterialCollection, 0)
```

Comment:
nit: just clean code.
```suggestion
	trustedMaterial := make(root.TrustedMaterialCollection)
```